pkg/velaql/view.go

```diff
@@ -131,6 +137,96 @@ func (handler *ViewHandler) delete(ctx context.Context, cluster string, owner co
 	return handler.cli.Delete(ctx, manifest)
 }
 
+// ValidateView makes sure the cue provided can use as view.
+//
+// For now, we only check 1. cue is valid 2. `status` field exists
+func ValidateView(viewStr string) error {
+	val, err := value.NewValue(viewStr, nil, "")
+	if err != nil {
+		return errors.Errorf("error when parsing view: %v", err)
+	}
+
+	// Make sure `status` field exists
+	v, err := val.LookupValue(DefaultExportValue)
+	if err != nil {
+		return errors.Errorf("no `status` field found in view: %v", err)
+	}
+	if _, err := v.String(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ParseViewIntoConfigMap parses a CUE string (representing a view) into a ConfigMap
+// ready to be stored into etcd.
+func ParseViewIntoConfigMap(viewStr, name string) (*v1.ConfigMap, error) {
```

Comment:
let's merge the function with this : https://github.com/kubevela/kubevela/commit/0c190ef32948561b577ccb0ca8432aee74bcfc80#diff-dfc3f6e3b7231b7b41cf82b8e561f3cbf8191d214886e21d798f287f693806dbR997